### DISCUSSION TITLE
[ERROR: No module named 'open_webui.apps.webui'] on Adding a New Function

### DISCUSSION THREAD

**Discussion by AzizDXT** [open-webui/open-webui#8119] [General]
#### **Title:**
[ERROR: No module named 'open_webui.apps.webui'] - Windows 11 / Python 3.11 / Open WebUI v0.5.1

---

#### **Description:**
When running Open WebUI (v0.5.1) on Windows 11 with Python 3.11 after adding a new function, the following error occurs:
```
[ERROR: No module named 'open_webui.apps.webui']
```

---

#### **Environment:**
- **OS:** Windows 11
- **Python Version:** 3.11
- **Open WebUI Version:** 0.5.1

---

#### **Steps to Reproduce:**
1. Install Open WebUI:
   ```bash
   pip install "C:\Users\abdul\Downloads\open-webui-0.5.1.tar.gz"
   ```
2. Add a new function to `apps/webui.py`:
   ```python
   def new_function():
       print("This is a new function!")
   ```
3. Run Open WebUI:
   ```bash
   python -m open_webui
   ```
4. Error appears:
   ```
   [ERROR: No module named 'open_webui.apps.webui']
   ```

---

#### **Expected Behavior:**
- Open WebUI should run without errors.

#### **Actual Behavior:**
- The program fails to run with the mentioned error.

---

#### **Notes:**
- The issue might be related to module paths or installation problems.

---

#### **Repository Link:**
[[Open WebUI GitHub Repository](https://github.com/open-webui/open-webui)](https://github.com/open-webui/open-webui)

---

This is a simplified bug report to submit to the "Issues" section on GitHub. Let me know if you need help submitting it!

**Comment by dannykorpan** [comment]
Same problem here, using 0.5.1.

**Comment by ghost** [comment]
Same problem here, using 0.5.1. Previously, everything worked fine with this tool. https://github.com/mhioi/open-webui-stuff/blob/met-tables/tools/MEET-tables/MEET-Tables.py     
[ERROR: No module named 'open_webui.apps']

**Comment by tjbck** [comment]
Please read the changelogs before updating, there has been a major refactoring with the backend and functions that utilise internal backend functions will not be compatible. I suggest you reach out to the function author.

Migration guide: https://docs.openwebui.com/features/plugin/migration/

**Reply by mindthemix** [reply]
Thanks man!

**Comment by ktxed** [comment]
some imports have been changed as @tjbck mentioned above, if you previously used somethig like:
```
from open_webui.apps.retrieval import main
from open_webui.apps.webui.models.knowledge import KnowledgeTable
```
now you must use:
```
from open_webui.routers.retrieval import QueryCollectionsForm, query_collection_handler
from open_webui.models.knowledge import KnowledgeTable
```
you can browse the current structure in order to replace your imports accordingly

**Comment by diegopedroso** [comment]
rollback to the `v0.4.0` solved my issue when i have tried to create new Pipeline functions 

https://github.com/open-webui/open-webui/releases/tag/v0.4.0

**Comment by joestump** [comment]
Just ran into this:

```
Error: No module named 'open_webui.apps'
```

Showed up when I ran a side-by-side of two models. I'm on v0.5.6.

**Comment by Wombosvideo** [comment]
I faced this issue when using code in my prompt. I fixed it by wrapping my code in tripple backticks:

````md
<prompt-prefix/>
```
<python-code/>
```
<prompt-suffix/>
````
